app/forms/idv/finance_form.rb

```diff
@@ -39,6 +39,13 @@ def finance_type_valid
       end
     end
 
+    def ccn_valid
+      return true unless finance_type == :ccn
+      unless finance_account =~ /^\d{8}$/
+        errors.add :finance_account, I18n.t('idv.errors.invalid_ccn')
+      end
+    end
```

Comment:
Would the built-in ActiveModel format validation work here?

```ruby
validates :finance_account, format: { with: /^\d{8}$/ }, if: finance_type == :ccn
```

For consistency and convention, I think we should use the built-in validations when possible. Some, like the uniqueness validation, are not supported by ActiveModel, so we have to write our own.